### DISCUSSION TITLE
Postal Codes

### DISCUSSION THREAD

**Discussion by jottinger** [tomaytotomato/location4j#31] [General]
Been wondering how postal codes would work - the problem is the formats for postal codes are so different, plus there are some regions that are too sparse to HAVE postal codes.

**Comment by tomaytotomato** [comment]
Having a look here - https://en.wikipedia.org/wiki/List_of_postal_codes

It looks like a majority of countries opt for numeric post/zipcodes. The problem then is what postcode is it?

There would need to be some context or guidance. 


```java

postcodeService.lookup("90210"); // many results?

postcodeService.lookup("90210", Country.USA); //beverly hills, california

```

### Alternative 

Build individual datasets of postcodes for countries and allow users to decide what relevant postcodes they want. 

Likelihood of someone needing Afghanisation postcodes vs. likelihood of someone needing postcodes of a developed country?

```java

String USA_POSTCODES = "/location4j-usa-postcodes.bin";
String UK_POSTCODES = "/location4j-uk-postcodes.bin";


SearchLocationService.builder().addData(UK_POSTCODES, USA_POSTCODES).build();
```
